File: EPLPredictorBackend/table.py
class TableEntry:
    def __init__(self, name, wins, draws, losses, goals_for, goals_conceded, points):
        self.name = name
        self.wins = wins
        self.draws = draws
        self.losses = losses
        self.goals_for = goals_for
        self.goals_conceded = goals_conceded
        self.goal_difference = goals_for - goals_conceded
        self.points = points


class Table:
    def __init__(self):
        self.leagueTable = []

    def add_result(self, name, result, goals_scored, goals_conceded):
        curr_team = [team for team in self.leagueTable if team.name == name]
        curr_team[0].goals_for += int(goals_scored)
        curr_team[0].goals_conceded += int(goals_conceded)
        curr_team[0].goal_difference += int(goals_scored) - int(goals_conceded)
        if result == 'W':
            curr_team[0].wins += 1
            curr_team[0].points += 3
        if result == 'D':
            curr_team[0].draws += 1
            curr_team[0].points += 1
        if result == 'L':
            curr_team[0].losses += 1

    def add_team(self, name, wins, draws, losses, goals_scored, goals_conceded, points):
        self.leagueTable.append(TableEntry(name, wins, draws, losses, goals_scored, goals_conceded, points))

    def get_table(self):
        return sorted(self.leagueTable, key=lambda e: (e.points, e.goal_difference), reverse=True)

File: EPLPredictorUI/eplpredictorui/src/components/table.js
import React from 'react'
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import './table.css';

class LeagueTable extends React.Component {

    constructor(props) {
        super(props);
        this.state = {table: []}
    }

    render() {

        console.log(this.state.table);
        return (
        <TableContainer component={Paper}>
        <Table stickyHeader style={{minWidth: 650, maxHeight: 400}}>
          <TableHead>
            <TableRow>
            <TableCell align="right">Position</TableCell>
              <TableCell align="left">Team</TableCell>
              <TableCell align="right">Played</TableCell>
              <TableCell align="right">Wins</TableCell>
              <TableCell align="right">Draws</TableCell>
              <TableCell align="right">Losses</TableCell>
              
              <TableCell align="right">Goals Scored</TableCell>
              <TableCell align="right">Goals Conceded</TableCell>
              <TableCell align="right">Goal Difference</TableCell>
              <TableCell align="right">Points</TableCell>
            </TableRow>
          </TableHead>
          <TableBody className="rows">
            {this.state.table.map((row, index) => (
              <TableRow  key = {row.name} >
                <TableCell align="right">{index+1}</TableCell>
                <TableCell align="left">{row.name}</TableCell>
                <TableCell align="right">38</TableCell>
                <TableCell align="right">{row.wins}</TableCell>
                <TableCell align="right">{row.draws}</TableCell>
                <TableCell align="right">{row.losses}</TableCell>
                
                <TableCell align="right">{row.goals_for}</TableCell>
                <TableCell align="right">{row.goals_conceded}</TableCell>
                <TableCell align="right">{row.goal_difference}</TableCell>
                <TableCell align="right">{row.points}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer> 
        )
    }

    

    componentDidMount() {
        fetch('http://127.0.0.1:5000/api/table')
            .then(res => res.json())
             .then(   (result) => {
                    this.setState({
                        table: result.table 
                    })
                    
                }
            )
    }
}

export default LeagueTable